src/main/kotlin/com/hibob/ex8-lambdas-exercise.kt

```diff
@@ -0,0 +1,119 @@
+package com.hibob
+
+import java.time.LocalDate
+
+/*
+1. Make the Runner.init function more readable using Lambda with Receiver function usage
+2. Initiate the variable movie in line 11 with the function createGoodMovie()
+3. Implement pretty print using buildString function
+4. Make SpidermanMovieProduceActions nullable (if not so yet) and make the relevant adjustments
+*/
+
+fun main() {
+    val movie = createGoodMovie() //SpidermanNoWayHome() // fun that would return the interface, it can be nullable
+    val runner = Runner(movie)
+    val success = runner.init()
+
+    printSuccessMessage(success)
+    println("Pretty print: ${movie.prettyPrint()}")
+//    println("Json: ${movie.toJson()}")
+}
+
+fun createGoodMovie() : SpidermanNoWayHome {
+    return SpidermanNoWayHome()
+}
+
+fun printSuccessMessage(success: Boolean) {
+    if (success) {
+        println("The movie was successful")
+    } else {
+        println("The movie failed")
+    }
+}
+
+interface SpidermanMovieProduceActions {
+    fun signTobeyMaguire()
+    fun signAndrew()
+    fun signTom()
+    fun getVillains()
+    fun isThereLockdown(): Boolean
+    fun publish():Boolean
+
+    val title: String
+    val airDate: LocalDate
+    val imdbRank: Double
+}
+
+class SpidermanNoWayHome() : SpidermanMovieProduceActions {
+
+    override val title: String = "Spiderman - No Way Home"
+    override val airDate: LocalDate = LocalDate.of(2021,12,16)
+    override val imdbRank: Double = 9.6
+
+    fun prettyPrint(): String {
+        return buildString { appendLine("Title: $title")
+            appendLine("date: $airDate")
+            appendLine("rank: $imdbRank")}
+    }
+    override fun signTobeyMaguire() {
+        //  Tobey signed!
```

Comment:
What about those empty implementations?

Reply:
I thought that it wasn't required by the 1-4 req, and reading it again, I'm not sure - should I add vars of Boolean type, that e.g. if signAndrew is called - than false -> true and next time the func would be called, it would return true instead of false?
getVillains, what should that method do? return random true\false if succeeded in capturing the villains?

Reply:
I know - sorry -it wasnt clear and I had to leave...
you all understood that - I mean the json function that was commented...
Here it is a very simple print...

Reply:
updated json usage, would like to know if a better approach  to code it exist...